src/test/java/jUnit/JsonParserTest.java

```diff
@@ -49,25 +56,28 @@ public void readFromFileTest() {
     void notSuchFileExceptionTest(String path) {
```

Comment:
I would suggest creating parametrized test here to double check file with a name we define from the very beginning is created with a correct extension (hint - make it exception test with dataset of incorrect extensions as per task requirement) instead of verifying non-existent files are created

Reply:
you may want to use parametrizedtest and valuesource of strings